pkg/version/version_parser.go

```diff
@@ -251,13 +257,20 @@ func ParseVersion(version string) (*ParsedSemVer, error) {
 	if err != nil {
 		return nil, fmt.Errorf("parsing patch version: %w", err)
 	}
+
+	var isIndependentRelease bool
+	if matched, err := regexp.MatchString(isIndependentReleaseFormat, matches[namedGroups["buildmetadata"]]); err == nil && matched {
+		isIndependentRelease = true
```

Comment:
I would rather have this to be done on demand in the `IsIndependentRelease()` function since we preserve both the original string and the build metadata part